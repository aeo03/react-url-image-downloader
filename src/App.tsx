import { useMemo } from 'react';

const App = () => {
  return useMemo(() =>
    <div className="container">
      <img width="100%" src="https://i.postimg.cc/9MFvz25j/movingtogether-primary-804744d5605a360dc6a2faca9183c51f.jpg" alt="" />
      <button>Download</button>
    </div>, []);
}

export default App;
