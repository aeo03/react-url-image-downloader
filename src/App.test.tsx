import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import App from './App';

configure({ adapter: new Adapter() });

describe('App component', function() {
  it('should test App component', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
   });

   it('should render App component', () => {
    expect(shallow(<App />).find('.container').length).toEqual(1);
   });

  it('should render img tag', function() {
    expect(shallow(<App />).find('img').length).toEqual(1);
  });

  it('should render button tag', function() {
    expect(shallow(<App />).find('button').length).toEqual(1);
  });

});