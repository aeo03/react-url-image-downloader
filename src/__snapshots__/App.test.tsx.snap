// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should test App component 1`] = `ShallowWrapper {}`;
